<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml" xmlns:epub="http://www.idpf.org/2007/ops">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="errno — Standard errno system symbols" />
<meta property="og:type" content="website" />
<meta property="og:url" content="https://docs.python.org/3/library/errno.xhtml" />
<meta property="og:site_name" content="Python documentation" />
<meta property="og:description" content="This module makes available standard errno system symbols. The value of each symbol is the corresponding integer value. The names and descriptions are borrowed from linux/include/errno.h, which sho..." />
<meta property="og:image" content="https://docs.python.org/3/_static/og-image.png" />
<meta property="og:image:alt" content="Python documentation" />
<meta name="description" content="This module makes available standard errno system symbols. The value of each symbol is the corresponding integer value. The names and descriptions are borrowed from linux/include/errno.h, which sho..." />
<meta property="og:image:width" content="200" />
<meta property="og:image:height" content="200" />
<meta name="theme-color" content="#3776ab" />

    <title>errno — Standard errno system symbols</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/epub.css" />
    
    <link rel="canonical" href="https://docs.python.org/3/library/errno.html" />
    
      
    

    
    <style>
      @media only screen {
        table.full-width-table {
            width: 100%;
        }
      }
    </style>
 

  </head><body>


    <div class="document">
      <div class="documentwrapper">
          <div class="body" role="main">
            
  <section id="module-errno">
<span id="errno-standard-errno-system-symbols"></span><h1><code class="xref py py-mod docutils literal notranslate"><span class="pre">errno</span></code> — Standard errno system symbols</h1>
<hr class="docutils" />
<p>This module makes available standard <code class="docutils literal notranslate"><span class="pre">errno</span></code> system symbols. The value of each
symbol is the corresponding integer value. The names and descriptions are
borrowed from <code class="file docutils literal notranslate"><span class="pre">linux/include/errno.h</span></code>, which should be
all-inclusive.</p>
<dl class="py data">
<dt class="sig sig-object py" id="errno.errorcode">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">errorcode</span></span></dt>
<dd><p>Dictionary providing a mapping from the errno value to the string name in the
underlying system.  For instance, <code class="docutils literal notranslate"><span class="pre">errno.errorcode[errno.EPERM]</span></code> maps to
<code class="docutils literal notranslate"><span class="pre">'EPERM'</span></code>.</p>
</dd></dl>

<p>To translate a numeric error code to an error message, use <a class="reference internal" href="os.xhtml#os.strerror" title="os.strerror"><code class="xref py py-func docutils literal notranslate"><span class="pre">os.strerror()</span></code></a>.</p>
<p>Of the following list, symbols that are not used on the current platform are not
defined by the module.  The specific list of defined symbols is available as
<code class="docutils literal notranslate"><span class="pre">errno.errorcode.keys()</span></code>.  Symbols available can include:</p>
<dl class="py data">
<dt class="sig sig-object py" id="errno.EPERM">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EPERM</span></span></dt>
<dd><p>Operation not permitted. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#PermissionError" title="PermissionError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">PermissionError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOENT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOENT</span></span></dt>
<dd><p>No such file or directory. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#FileNotFoundError" title="FileNotFoundError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">FileNotFoundError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ESRCH">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ESRCH</span></span></dt>
<dd><p>No such process. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#ProcessLookupError" title="ProcessLookupError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ProcessLookupError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EINTR">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EINTR</span></span></dt>
<dd><p>Interrupted system call. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#InterruptedError" title="InterruptedError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">InterruptedError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EIO">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EIO</span></span></dt>
<dd><p>I/O error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENXIO">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENXIO</span></span></dt>
<dd><p>No such device or address</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.E2BIG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">E2BIG</span></span></dt>
<dd><p>Arg list too long</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOEXEC">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOEXEC</span></span></dt>
<dd><p>Exec format error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBADF">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBADF</span></span></dt>
<dd><p>Bad file number</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ECHILD">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ECHILD</span></span></dt>
<dd><p>No child processes. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#ChildProcessError" title="ChildProcessError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ChildProcessError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EAGAIN">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EAGAIN</span></span></dt>
<dd><p>Try again. This error is mapped to the exception <a class="reference internal" href="exceptions.xhtml#BlockingIOError" title="BlockingIOError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">BlockingIOError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOMEM">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOMEM</span></span></dt>
<dd><p>Out of memory</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EACCES">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EACCES</span></span></dt>
<dd><p>Permission denied.  This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#PermissionError" title="PermissionError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">PermissionError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EFAULT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EFAULT</span></span></dt>
<dd><p>Bad address</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTBLK">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTBLK</span></span></dt>
<dd><p>Block device required</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBUSY">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBUSY</span></span></dt>
<dd><p>Device or resource busy</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EEXIST">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EEXIST</span></span></dt>
<dd><p>File exists. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#FileExistsError" title="FileExistsError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">FileExistsError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EXDEV">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EXDEV</span></span></dt>
<dd><p>Cross-device link</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENODEV">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENODEV</span></span></dt>
<dd><p>No such device</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTDIR">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTDIR</span></span></dt>
<dd><p>Not a directory. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#NotADirectoryError" title="NotADirectoryError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">NotADirectoryError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EISDIR">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EISDIR</span></span></dt>
<dd><p>Is a directory. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#IsADirectoryError" title="IsADirectoryError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">IsADirectoryError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EINVAL">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EINVAL</span></span></dt>
<dd><p>Invalid argument</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENFILE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENFILE</span></span></dt>
<dd><p>File table overflow</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EMFILE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EMFILE</span></span></dt>
<dd><p>Too many open files</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTTY">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTTY</span></span></dt>
<dd><p>Not a typewriter</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ETXTBSY">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ETXTBSY</span></span></dt>
<dd><p>Text file busy</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EFBIG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EFBIG</span></span></dt>
<dd><p>File too large</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOSPC">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOSPC</span></span></dt>
<dd><p>No space left on device</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ESPIPE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ESPIPE</span></span></dt>
<dd><p>Illegal seek</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EROFS">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EROFS</span></span></dt>
<dd><p>Read-only file system</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EMLINK">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EMLINK</span></span></dt>
<dd><p>Too many links</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EPIPE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EPIPE</span></span></dt>
<dd><p>Broken pipe. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#BrokenPipeError" title="BrokenPipeError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">BrokenPipeError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EDOM">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EDOM</span></span></dt>
<dd><p>Math argument out of domain of func</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ERANGE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ERANGE</span></span></dt>
<dd><p>Math result not representable</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EDEADLK">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EDEADLK</span></span></dt>
<dd><p>Resource deadlock would occur</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENAMETOOLONG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENAMETOOLONG</span></span></dt>
<dd><p>File name too long</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOLCK">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOLCK</span></span></dt>
<dd><p>No record locks available</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOSYS">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOSYS</span></span></dt>
<dd><p>Function not implemented</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTEMPTY">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTEMPTY</span></span></dt>
<dd><p>Directory not empty</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ELOOP">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ELOOP</span></span></dt>
<dd><p>Too many symbolic links encountered</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EWOULDBLOCK">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EWOULDBLOCK</span></span></dt>
<dd><p>Operation would block. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#BlockingIOError" title="BlockingIOError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">BlockingIOError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOMSG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOMSG</span></span></dt>
<dd><p>No message of desired type</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EIDRM">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EIDRM</span></span></dt>
<dd><p>Identifier removed</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ECHRNG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ECHRNG</span></span></dt>
<dd><p>Channel number out of range</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EL2NSYNC">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EL2NSYNC</span></span></dt>
<dd><p>Level 2 not synchronized</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EL3HLT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EL3HLT</span></span></dt>
<dd><p>Level 3 halted</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EL3RST">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EL3RST</span></span></dt>
<dd><p>Level 3 reset</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ELNRNG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ELNRNG</span></span></dt>
<dd><p>Link number out of range</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EUNATCH">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EUNATCH</span></span></dt>
<dd><p>Protocol driver not attached</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOCSI">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOCSI</span></span></dt>
<dd><p>No CSI structure available</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EL2HLT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EL2HLT</span></span></dt>
<dd><p>Level 2 halted</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBADE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBADE</span></span></dt>
<dd><p>Invalid exchange</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBADR">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBADR</span></span></dt>
<dd><p>Invalid request descriptor</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EXFULL">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EXFULL</span></span></dt>
<dd><p>Exchange full</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOANO">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOANO</span></span></dt>
<dd><p>No anode</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBADRQC">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBADRQC</span></span></dt>
<dd><p>Invalid request code</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBADSLT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBADSLT</span></span></dt>
<dd><p>Invalid slot</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EDEADLOCK">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EDEADLOCK</span></span></dt>
<dd><p>File locking deadlock error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBFONT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBFONT</span></span></dt>
<dd><p>Bad font file format</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOSTR">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOSTR</span></span></dt>
<dd><p>Device not a stream</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENODATA">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENODATA</span></span></dt>
<dd><p>No data available</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ETIME">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ETIME</span></span></dt>
<dd><p>Timer expired</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOSR">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOSR</span></span></dt>
<dd><p>Out of streams resources</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENONET">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENONET</span></span></dt>
<dd><p>Machine is not on the network</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOPKG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOPKG</span></span></dt>
<dd><p>Package not installed</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EREMOTE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EREMOTE</span></span></dt>
<dd><p>Object is remote</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOLINK">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOLINK</span></span></dt>
<dd><p>Link has been severed</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EADV">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EADV</span></span></dt>
<dd><p>Advertise error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ESRMNT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ESRMNT</span></span></dt>
<dd><p>Srmount error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ECOMM">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ECOMM</span></span></dt>
<dd><p>Communication error on send</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EPROTO">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EPROTO</span></span></dt>
<dd><p>Protocol error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EMULTIHOP">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EMULTIHOP</span></span></dt>
<dd><p>Multihop attempted</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EDOTDOT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EDOTDOT</span></span></dt>
<dd><p>RFS specific error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBADMSG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBADMSG</span></span></dt>
<dd><p>Not a data message</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EOVERFLOW">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EOVERFLOW</span></span></dt>
<dd><p>Value too large for defined data type</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTUNIQ">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTUNIQ</span></span></dt>
<dd><p>Name not unique on network</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EBADFD">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EBADFD</span></span></dt>
<dd><p>File descriptor in bad state</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EREMCHG">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EREMCHG</span></span></dt>
<dd><p>Remote address changed</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ELIBACC">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ELIBACC</span></span></dt>
<dd><p>Can not access a needed shared library</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ELIBBAD">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ELIBBAD</span></span></dt>
<dd><p>Accessing a corrupted shared library</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ELIBSCN">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ELIBSCN</span></span></dt>
<dd><p>.lib section in a.out corrupted</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ELIBMAX">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ELIBMAX</span></span></dt>
<dd><p>Attempting to link in too many shared libraries</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ELIBEXEC">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ELIBEXEC</span></span></dt>
<dd><p>Cannot exec a shared library directly</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EILSEQ">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EILSEQ</span></span></dt>
<dd><p>Illegal byte sequence</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ERESTART">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ERESTART</span></span></dt>
<dd><p>Interrupted system call should be restarted</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ESTRPIPE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ESTRPIPE</span></span></dt>
<dd><p>Streams pipe error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EUSERS">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EUSERS</span></span></dt>
<dd><p>Too many users</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTSOCK">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTSOCK</span></span></dt>
<dd><p>Socket operation on non-socket</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EDESTADDRREQ">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EDESTADDRREQ</span></span></dt>
<dd><p>Destination address required</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EMSGSIZE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EMSGSIZE</span></span></dt>
<dd><p>Message too long</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EPROTOTYPE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EPROTOTYPE</span></span></dt>
<dd><p>Protocol wrong type for socket</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOPROTOOPT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOPROTOOPT</span></span></dt>
<dd><p>Protocol not available</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EPROTONOSUPPORT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EPROTONOSUPPORT</span></span></dt>
<dd><p>Protocol not supported</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ESOCKTNOSUPPORT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ESOCKTNOSUPPORT</span></span></dt>
<dd><p>Socket type not supported</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EOPNOTSUPP">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EOPNOTSUPP</span></span></dt>
<dd><p>Operation not supported on transport endpoint</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTSUP">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTSUP</span></span></dt>
<dd><p>Operation not supported</p>
<div class="versionadded">
<p><span class="versionmodified added">Added in version 3.2.</span></p>
</div>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EPFNOSUPPORT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EPFNOSUPPORT</span></span></dt>
<dd><p>Protocol family not supported</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EAFNOSUPPORT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EAFNOSUPPORT</span></span></dt>
<dd><p>Address family not supported by protocol</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EADDRINUSE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EADDRINUSE</span></span></dt>
<dd><p>Address already in use</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EADDRNOTAVAIL">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EADDRNOTAVAIL</span></span></dt>
<dd><p>Cannot assign requested address</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENETDOWN">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENETDOWN</span></span></dt>
<dd><p>Network is down</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENETUNREACH">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENETUNREACH</span></span></dt>
<dd><p>Network is unreachable</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENETRESET">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENETRESET</span></span></dt>
<dd><p>Network dropped connection because of reset</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ECONNABORTED">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ECONNABORTED</span></span></dt>
<dd><p>Software caused connection abort. This error is mapped to the
exception <a class="reference internal" href="exceptions.xhtml#ConnectionAbortedError" title="ConnectionAbortedError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ConnectionAbortedError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ECONNRESET">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ECONNRESET</span></span></dt>
<dd><p>Connection reset by peer. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#ConnectionResetError" title="ConnectionResetError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ConnectionResetError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOBUFS">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOBUFS</span></span></dt>
<dd><p>No buffer space available</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EISCONN">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EISCONN</span></span></dt>
<dd><p>Transport endpoint is already connected</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTCONN">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTCONN</span></span></dt>
<dd><p>Transport endpoint is not connected</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ESHUTDOWN">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ESHUTDOWN</span></span></dt>
<dd><p>Cannot send after transport endpoint shutdown. This error is mapped
to the exception <a class="reference internal" href="exceptions.xhtml#BrokenPipeError" title="BrokenPipeError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">BrokenPipeError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ETOOMANYREFS">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ETOOMANYREFS</span></span></dt>
<dd><p>Too many references: cannot splice</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ETIMEDOUT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ETIMEDOUT</span></span></dt>
<dd><p>Connection timed out. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#TimeoutError" title="TimeoutError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">TimeoutError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ECONNREFUSED">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ECONNREFUSED</span></span></dt>
<dd><p>Connection refused. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#ConnectionRefusedError" title="ConnectionRefusedError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">ConnectionRefusedError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EHOSTDOWN">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EHOSTDOWN</span></span></dt>
<dd><p>Host is down</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EHOSTUNREACH">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EHOSTUNREACH</span></span></dt>
<dd><p>No route to host</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EALREADY">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EALREADY</span></span></dt>
<dd><p>Operation already in progress. This error is mapped to the
exception <a class="reference internal" href="exceptions.xhtml#BlockingIOError" title="BlockingIOError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">BlockingIOError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EINPROGRESS">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EINPROGRESS</span></span></dt>
<dd><p>Operation now in progress. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#BlockingIOError" title="BlockingIOError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">BlockingIOError</span></code></a>.</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ESTALE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ESTALE</span></span></dt>
<dd><p>Stale NFS file handle</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EUCLEAN">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EUCLEAN</span></span></dt>
<dd><p>Structure needs cleaning</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTNAM">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTNAM</span></span></dt>
<dd><p>Not a XENIX named type file</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENAVAIL">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENAVAIL</span></span></dt>
<dd><p>No XENIX semaphores available</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EISNAM">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EISNAM</span></span></dt>
<dd><p>Is a named type file</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EREMOTEIO">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EREMOTEIO</span></span></dt>
<dd><p>Remote I/O error</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EDQUOT">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EDQUOT</span></span></dt>
<dd><p>Quota exceeded</p>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EQFULL">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EQFULL</span></span></dt>
<dd><p>Interface output queue is full</p>
<div class="versionadded">
<p><span class="versionmodified added">Added in version 3.11.</span></p>
</div>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTCAPABLE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTCAPABLE</span></span></dt>
<dd><p>Capabilities insufficient. This error is mapped to the exception
<a class="reference internal" href="exceptions.xhtml#PermissionError" title="PermissionError"><code class="xref py py-exc docutils literal notranslate"><span class="pre">PermissionError</span></code></a>.</p>
<div class="availability docutils container">
<p><a class="reference internal" href="intro.xhtml#availability"><span class="std std-ref">Availability</span></a>: WASI, FreeBSD</p>
</div>
<div class="versionadded">
<p><span class="versionmodified added">Added in version 3.11.1.</span></p>
</div>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ECANCELED">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ECANCELED</span></span></dt>
<dd><p>Operation canceled</p>
<div class="versionadded">
<p><span class="versionmodified added">Added in version 3.2.</span></p>
</div>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.EOWNERDEAD">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">EOWNERDEAD</span></span></dt>
<dd><p>Owner died</p>
<div class="versionadded">
<p><span class="versionmodified added">Added in version 3.2.</span></p>
</div>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="errno.ENOTRECOVERABLE">
<span class="sig-prename descclassname"><span class="pre">errno.</span></span><span class="sig-name descname"><span class="pre">ENOTRECOVERABLE</span></span></dt>
<dd><p>State not recoverable</p>
<div class="versionadded">
<p><span class="versionmodified added">Added in version 3.2.</span></p>
</div>
</dd></dl>

</section>


            <div class="clearer"></div>
          </div>
      </div>
      <div class="clearer"></div>
    </div>
  </body>
</html>